extension/js/popup.js

```diff
@@ -65,6 +73,15 @@
             }
         });
 
+        SiteProfile.retreive(null, function (profiles) {
+            profiles.forEach(function (profile, i) {
+                if (profile.domain === $('#domain').val()) {
+                    siteProfile = profile;
+                    updateDomain();
+                }
+            });
+        });
+
```

Comment:
Couple of points with this:

 - This could be pretty slow with lots of domains. `$('#domain').val()` should be cached before the `forEach()` to make this comparison faster. 
 - Also, rather than using `forEach`, consider using [`some`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/some) which will break on the first hit. 
 - Rather than having the `siteProfile` var and running `updateDomain` as the side-effect, why not pass `siteProfile` to `updateDomain`?

I would refactor this code with the above points, making it look like:

```js
SiteProfile.retreive(null, function (profiles) {
    var domain = $('#domain').val();
    profiles.some(function (profile) {
        if (profile.domain === domain) {
            updateDomain(profile);
            return true;
        }
    });
});
```